libs/staging-provider/src/Provider/BaseWorkspaceProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StagingProvider\Provider;
+
+use Keboola\StagingProvider\Exception\StagingProviderException;
+use Keboola\StorageApi\Workspaces;
+
+abstract class BaseWorkspaceProvider implements WorkspaceProviderInterface
+{
+    public function __construct(
+        private readonly Workspaces $workspacesApiClient,
+    ) {
+    }
+
+    abstract protected function getWorkspace(): Workspace;
+
+    public function getBackendSize(): ?string
+    {
+        return $this->getWorkspace()->getBackendSize();
+    }
+
+    public function getBackendType(): string
+    {
+        return $this->getWorkspace()->getBackendType();
+    }
+
+    public function getPath(): string
+    {
+        throw new StagingProviderException(sprintf(
+            '%s does not support path',
+            static::class,
+        ));
+    }
+
+    public function cleanup(): void
+    {
+        $this->workspacesApiClient->deleteWorkspace((int) $this->getWorkspaceId(), [], true);
```

Comment:
jaktože to nesmaže workspace v existing workspace provideru? tam ta metoda není přetížená
https://github.com/keboola/platform-libraries/pull/390/files#diff-7a2bdacd4deb8b701412869520fc4abc53bd16d64ce6a45fb570b14aa1639d90R50

Reply:
Neco mi unika? Normalne se to smazne
* v `ExistingWorkspaceProvider` neni metoda pretizena = pouzije se tahle = proste to smazne workspace
* v `NewWorkspaceProvider` je pretizena, aby se `parent` nevolal, pokud workspace nebyl zalozeny

Reply:
nj, ono je to řešený tímhle https://github.com/keboola/docker-bundle/pull/785/files#diff-572fc133f255533867b23e1f8cb4e9e73d68b726b2d5254ddd19904ff1eb09bbL450
 to jako je skoro na to aby ten cleanup v tom ExistingWorkspaceProvideru vyhodil vyjimku, protože to prostě zavolat nechceš (a rozhodně nechceš po WS provideru, aby ti smazal WS kterej jsi mu vložil zvenku). Ale bylo to tak, může to tak zůstat